### PR TITLE
Partially implement LTC Backend interface for PyTorch/XLA.

### PR DESCRIPTION
**Pull request by JackCaoG**
Partially implement LTC Backend interface for PyTorch/XLA.

### CODE REVIEW

**Comment by JackCaoG** [comment]
@wonjoolee95 I think this pr is ready to be review. This pr only partially implement LTC Backend interface. I will have follow up pr(s) for the remaining prs.

File: torch_xla/csrc/xla_backend_impl.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  // GetCompilationDevices always returns devices today because
+  // 1. devices passed to GetCompilationDevices is always empty
+  // 2. ComputationClient's replication device is never set since we always have
+  // one replica per process
+  // We should revisit this API for SPMD
+  std::vector<std::string> GetCompilationDevices(
+      const std::string& device,
+      c10::ArrayRef<std::string> devices) const override {
+    return xla::ComputationClient::Get()->GetCompilationDevices(device,
```
Comment (JackCaoG) [inline torch_xla/csrc/xla_backend_impl.cpp:89]: fyi @yeounoh, we currently call it in https://github.com/pytorch/xla/blob/185dd3cd45e4a1b9dcbbf88a3c604ce8c5e752f3/torch_xla/csrc/tensor.cpp#L1666.

**Review by wonjoolee95 — APPROVED** [review]
Thanks! So this `XlaBackendImpl` won't be used yet in XLA until we implement the `InitXlaBackend` function, correct?

**Comment by JackCaoG** [comment]
@wonjoolee95 It won't be used even if we implement `InitXlaBackend` for the most part. Some functions needs to call `getBackend()->xxxx` to execute the function we defined here. Currently most of the `Backend` function is used the the ltc's LazyGraphExecutor.cpp. Until we fully migrated to that this file won't be very relevant.

One thing we can do incrementally is to start using the function in `XlaBackend` in our `torch_xla/csrc/tensor.cpp`. That way we can get some early signal of the correctness of these methods.

**Pull request merged**